xla/service/cpu/onednn_convolution.cc

```diff
@@ -208,6 +210,167 @@ CreateOneDnnPrimDesc<dnnl::convolution_forward::primitive_desc>(
       any_res_md, strides, rhs_dilations, pad_left, pad_right, attrs);
 }
 
+void ExecuteOneDnnConvolution(void* result, void* scratch, void** args,
```

Comment:
The new function should use the new FFI interface instead of using `void*` pointers.

Reply:
JAX has plenty of examples in the example and jaxlib folder:

* https://github.com/jax-ml/jax/tree/main/examples/ffi
* https://github.com/jax-ml/jax/blob/main/jaxlib/cpu/lapack_kernels.cc
* https://github.com/jax-ml/jax/blob/main/jaxlib/cpu/sparse_kernels.cc
* https://github.com/jax-ml/jax/blob/main/jaxlib/gpu/hybrid_kernels.cc

For opaque types, you can just use `ffi::AnyBuffer`.